client/js/compute.js

```diff
@@ -0,0 +1,578 @@
+    {
+      name: 'trimEnd',
+      desc: 'returns the string after removing withspace from the end',
+      sample: 'var a = ${x} trimEnd',
+      call: function(v, x) { return v = x.trimEnd() }
+    },
+    {
+      name: 'charAt',
+      desc: 'returns the nth character of a string x, with 0 being the first',
+      sample: 'var a = ${x} charAt ${n}',
+      call: function(v, x, y) { return v = (y !== undefined) ? x.charAt(y) : x.charAt() }
+    },
+    {
+      name: 'charCodeAt',
+      desc: 'returns the Unicode of the nth character in a string x, with 0 being the first',
+      sample: 'var a = ${x} charCodeAt ${n}',
+      call: function(v, x, y) { return v = (y !== undefined) ? x.charCodeAt(y) : x.charCodeAt() }
+    },
+    {
+      name: 'codePointAt',
+      desc: 'returns the UTF-16 code point value of the nth character in a string x, with 0 being the first',
+      sample: 'var a = ${x} codePointAt ${n}',
+      call: function(v, x, y) { return v = (y !== undefined) ? x.codePointAt(y) : x.codePointAt() }
+    },
+    {
+      name: 'concat',
+      desc: 'returns a string containing the text of the joined strings',
+      sample: 'var a = ${x} concat ${y}',
+      call: function(v, x, y) { return v = x.concat(y) }
+    },
+    {
+      name: 'includes',
+      desc: 'returns true if a string x contains a string y - case sensitive',
+      sample: 'var a = ${x} includes ${y}',
+      call: function(v, x, y) { return v = x.includes(y) }
+    },
+    {
+      name: 'endsWith',
+      desc: 'returns true if a string x ends with a string y - case sensitive',
+      sample: 'var a = ${x} endsWith ${y}',
+      call: function(v, x, y) { return v = x.endsWith(y) }
+    },
+    {
+      name: 'indexOf',
+      desc: 'returns the position of the first occurrence of a string y in string x, or -1 if not found - case sensitive',
+      sample: 'var a = ${x} indexOf ${y}',
+      call: function(v, x, y) { return v = x.indexOf(y) }
+    },
+    {
+      name: 'lastIndexOf',
+      desc: 'returns the position of the last occurrence of a string y in string x, or -1 if not found - case sensitive',
+      sample: 'var a = ${x} lastIndexOf ${y}',
+      call: function(v, x, y) { return v = x.lastIndexOf(y) }
+    },
+    {
+      name: 'in',
+      desc: 'returns true if the string x is included in array y (or property x in object y) - case sensitive',
+      sample: 'var a = ${x} in ${y}',
+      call: function(v, x, y) { return v = Array.isArray(y) ? y.indexOf(x) != -1 : x in y }
+    },
+    {
+      name: 'localeCompare',
+      desc: 'compares the sort order of two strings and returns -1 (x before y), 0 (x equals y) or 1 (x after y) - case sensitive',
+      sample: 'var a = ${x} localeCompare ${y}',
+      call: function(v, x, y) { return v = x.localeCompare(y) }
+    },
+    {
+      name: 'match',
+      desc: 'searches a string x for a match against a regular expression, and returns the matches as an Array, or null if no match was found',
```

Comment:
Any reason for the capital "Array"? Or just German being German?

Do these even work if the regex is a string? (Not related to the PR of course.)

Reply:
A regex string DOES work, but match only returns the first match.  Presumably, because the /g global modifier is absent.

Reply:
I've added a `z` argument which allows to use flags like 'g' for the global modifier. These kind of optional arguments are still a pain because of our default values, so I'm now assuming that z is not present if its value is 1 (the default).